src/ucs/profile/profile.c

```diff
@@ -253,22 +300,43 @@ static void ucs_profile_write(ucs_profile_context_t *ctx)
     header.num_locations = ctx->num_locations;
     header.num_threads   = ucs_list_length(&ctx->thread_list);
     header.one_second    = ucs_time_from_sec(1.0);
-    ucs_profile_file_write_data(fd, &header, sizeof(header));
+    header.feature_flags = UCS_PROFILE_ENV_VARIABLES;
+
+    /* Leave empty space for the header to be written later (due to offsets and sizes calculations) */
+    if (lseek(fd, sizeof(header), SEEK_SET) < 0) {
+        goto out_close_fd;
+    }
```

Comment:
why can't we write in sequentially?

Reply:
Because the profile_block_header structs contains info that calculated while writing the rest of the file.
This might be avoided if we will replace the use of lseek in the macro with struct size calculations

Reply:
yeah, i think we can calc the sizes w/o lseek